### PR TITLE
docs: misc update

### LINKED ISSUE #44027: 在Form.item中，使用ImgCrop包裹下的Upload 无法触发validator方法
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/zhu-ce-xin-yong-hu-antd-5-8-2-forked-6rr62h?file=/demo.tsx)

### Steps to reproduce

点击删除图片，rules未出现提示

### What is expected?

能够触发validator

### What is actually happening?

未能触发validator方法里面的逻辑

| Environment | Info |
| --- | --- |
| antd | undefined |
| React | 18.0 |
| System | Win11 |
| Browser | EDGE |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #44027

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |    -       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 012582c</samp>

This pull request updates the documentation and the post-script for the Form component, to address a bug with Form.Item and nested components. It adds a new section to the `components/form/index.en-US.md` and `components/form/index.zh-CN.md` files, explaining the cause and the solution of the bug, and a new entry to the `scripts/post-script.js` file, warning users of the deprecated version 5.8.0 and linking to issue #43943.

### <samp>🤖 Generated by Copilot at 012582c</samp>

* Add a new section to the Form component documentation, explaining how to use HOCs to handle nested children components in Form.Item ([link](https://github.com/ant-design/ant-design/pull/44034/files?diff=unified&w=0#diff-28baad7eca32a14feb2ed5fdcee826cb1d27ee818bcbcaa6c9a8ed368c9b16e5R649-R667), [link](https://github.com/ant-design/ant-design/pull/44034/files?diff=unified&w=0#diff-f17ef6685c42eb193b5735088ef3c54e36aca982b8464077915301cabb1d4b8fR649-R665))
* Update the post-script.js file to warn users of the bug with Form.Item and nested components, and link to the issue and the solution ([link](https://github.com/ant-design/ant-design/pull/44034/files?diff=unified&w=0#diff-0c429b9a01f20777bce85b2ad2472fe50f00b4ef8a404d56852f3d4dd15e905dR47))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

File: components/form/index.en-US.md [hunk 1]
```diff
@@ -646,3 +646,22 @@ Form can not get real DOM node when customize component not support `ref`. It wi
 ### `setFieldsValue` do not trigger `onFieldsChange` or `onValuesChange`?
 
 It's by design. Only user interactive can trigger the change event. This design is aim to avoid call `setFieldsValue` in change event which may makes loop calling.
+
+### 为什么 Form.Item 嵌套子组件后，更新表单值？
```
Comment (kiner-tang) [inline components/form/index.en-US.md:650]: 这个要删掉吧

File: components/form/index.zh-CN.md [hunk 2]
```diff
@@ -646,6 +646,23 @@ React 中异步更新会导致受控组件交互行为异常。当用户交互
 
 是的，change 事件仅当用户交互才会触发。该设计是为了防止在 change 事件中调用 `setFieldsValue` 导致的循环问题。如果仅仅需要组件内消费，可以通过 `useWatch` 或者 `Field.renderProps` 来实现。
 
+### 为什么 Form.Item 嵌套子组件后，更新表单值？
```
Comment (linxianxi) [inline components/form/index.zh-CN.md:649]: 病句。是不是改成 Form.Item 嵌套子组件后，如何更新表单值？

Comment (linxianxi) [reply components/form/index.zh-CN.md:649]: 或者：为什么 Form.Item 嵌套子组件后，不更新表单值？

Comment (kiner-tang) [reply components/form/index.zh-CN.md:649]: 让 GPT 润色一下，瞬间去除病句，哈哈

File: components/form/index.zh-CN.md [hunk 3]
```diff
@@ -646,6 +646,23 @@ React 中异步更新会导致受控组件交互行为异常。当用户交互
 
 是的，change 事件仅当用户交互才会触发。该设计是为了防止在 change 事件中调用 `setFieldsValue` 导致的循环问题。如果仅仅需要组件内消费，可以通过 `useWatch` 或者 `Field.renderProps` 来实现。
 
+### 为什么 Form.Item 关联的字段组件在嵌套下会失效？
```
Comment (kiner-tang) [inline components/form/index.zh-CN.md:649]:
```suggestion
### 为什么 Form.Item 关联的字段组件在嵌套使用时会失效？
```

File: components/form/index.zh-CN.md [hunk 4]
```diff
@@ -646,6 +646,23 @@ React 中异步更新会导致受控组件交互行为异常。当用户交互
...
+
+Form.Item 在渲染时会注入 `value` 与 `onChange` 事件给子元素，当你的字段组件被包裹时属性将无法传递。你可以通过 HOC 自定义组件形式来解决这个问题：
```
Comment (afc163) [inline components/form/index.zh-CN.md:651]: > 当你的字段组件被包裹时属性将无法传递

先还原一下问题代码，这样用户才能对号入座。

File: components/form/index.en-US.md [hunk 5]
```diff
@@ -646,3 +646,31 @@ Form can not get real DOM node when customize component not support `ref`. It wi
...
+```jsx
+<Form.Item name="input">
+  <div>
+    <h3>I am a wrapped Input</h3>
+    <Input />
+  </div>
+</Form.Item>
+```
+
+You can use HOC to solve this problem:
```
Comment (afc163) [inline components/form/index.en-US.md:663]:
```suggestion
You can use HOC to solve this problem, don't forget passing props to form control component:
```

**Pull request merged**